paper/paper.Rmd

```diff
@@ -24,140 +25,735 @@ knitr::opts_chunk$set(echo = TRUE)
+
+```{r state-vis, fig.width= 6, fig.height=4}
+library(ggplot2)
+plot_state <- fit1$plot(state_estimation, additional_stats = "none") +
+  ggtitle("Probability of voting the BP by state") +
+  theme_bw()
+plot_state
+```
+
+Lastly, we show the plot of the population estimate, with lines representing the mean weighted and raw estimates. About 70 per cent of people in the Shape World will be likely to vote for the Box Party.
+
+```{r,popn_estimate, fig.width= 6, fig.height=4}
+fit1$plot(popn_estimation, additional_stats = c("raw","wtd"))
+```
+
+
+
+
+
+
+
+
+# Worked example {#example}
+
+In this example we will forecast the 2020 US presidential election using survey data from the Democracy Fund Voter Study Group and use the Integrated Public Use Microdata Series (IPUMS), to access the 2018 American Community Survey (ACS) to use as post-stratification data. We are interested in considering the following variables: state, age-group, and gender.
+
+The first step is to gather the datasets. For the survey data we are interested in the Nationscape dataset, which can be requested from the Voter Study Group. Access is usually granted within hours and is completely free. We are going to use the survey from 25 June 2020.
+
+```{r, warning=FALSE, message=FALSE}
+# This code based on that from Alen Mitrovski, Xiaoyan Yang, and Matthew Wankiewicz.
+library(haven)
+library(tidyverse)
+
+raw_nationscape_data <-
+  read_dta(here::here("paper/ns20200625.dta"))
+
+# The Stata format separates labels so reunite those
+raw_nationscape_data <-
+  labelled::to_factor(raw_nationscape_data)
+
+# Just keep relevant variables
+nationscape_data <-
+  raw_nationscape_data %>%
+  select(vote_2020,
+         gender,
+         state,
+         age)
+
+# For simplicity, remove anyone undecided or planning to vote for someone other than Biden/Trump and make vote a binary variable: 1 for Biden, 0 for Trump.
+nationscape_data <-
+  nationscape_data %>%
+  filter(vote_2020 == "Joe Biden" | vote_2020 == "Donald Trump") %>%
+  mutate(vote_biden = if_else(vote_2020 == "Joe Biden", 1, 0)) %>%
+  select(-vote_2020)
+
+# Create the dependent variables by grouping the existing variables
+nationscape_data <-
+  nationscape_data %>%
+  mutate(
+    age_group = case_when(
+      age <= 29 ~ 'age_18-29',
+      age <= 44 ~ 'age_30-44',
+      age <= 59 ~ 'age_45-59',
+      age >= 60 ~ 'age_60_or_more',
+      TRUE ~ 'Trouble'
+      )
+    )
+
```

Comment:
Just to clarify, is 29 being recoded to 18-29? What is the existing coding? Could you do this recoding in the package itself?